### PR TITLE
FIXED Delete empty item bug and new language text editing bug

### PR DESCRIPTION
**Pull request by eorahil**
related in issues #424  and maybe #374 

Deleting elements shows empty buttons that, if pressed, deletes all entries in the dictionary.

Fixed an issue that causes new language texts to be removed when saving

### CODE REVIEW

File: unity/Assets/Scripts/QuestEditor/EditorSelectionList.cs [hunk 1]
```diff
@@ -152,6 +154,11 @@ public void SelectItem(UnityEngine.Events.UnityAction call)
                 tb = new TextButton(new Vector2(21, offset), new Vector2(20, 1), 
                     new StringKey(null, filtered[i].name, false), delegate { SelectComponent(key); }, filtered[i].color);
                 tb.button.GetComponent<UnityEngine.UI.Text>().fontSize = UIScaler.GetSmallFont();
+                if (key == null || key.Length == 0)
```
Comment (NPBruce) [inline unity/Assets/Scripts/QuestEditor/EditorSelectionList.cs]: Length of 0 is used intentionally in many cases

**Comment by NPBruce** [comment]
I have done a fix on master and merged back, so this is now in conflict.

File: unity/Assets/Scripts/QuestEditor/EditorSelectionList.cs (hunk 1 above)
Comment (eorahil) [inline unity/Assets/Scripts/QuestEditor/EditorSelectionList.cs]: And there cases can't be replaced by " " with lenght 1?

Comment (NPBruce) [inline unity/Assets/Scripts/QuestEditor/EditorSelectionList.cs]: Yes, but they aren't currently. I'd prefer to replace with something like {none} or {empty}. I have added the check into the delete function anyway for this particular bug.

Comment (eorahil) [inline unity/Assets/Scripts/QuestEditor/EditorSelectionList.cs]: Ok so I should remote this condition only?

Comment (NPBruce) [inline unity/Assets/Scripts/QuestEditor/EditorSelectionList.cs]: Either that or review everywhere "" is used, I don't mind.

**Comment by eorahil** [comment]
Reviewed where "" used. I decided to change the 2 remove selectionItems with null and keep the others as "".

**Pull request merged**